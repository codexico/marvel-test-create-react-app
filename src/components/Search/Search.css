

.Search {
  width: 70%;
  --height: 4rem;
  margin: 0 auto;
  padding: 12px;

  border-radius: calc(var(--height) / 2);
  background-color: var(--color-wisp-pink);
}

.Search-label {
  display: flex;
}

.Search-inputButton {
  margin-right: 20px;

  border: none;
  background: none;
}

.Search-inputIcon {
  width: auto;
  height: calc(var(--height) / 2);
}

.Search-input {
  flex-grow: 1;

  height: calc(var(--height) - 12px);

  border: none;
  background: none;

  color: var(--font-color-input);
  font-size: calc(var(--height) / 2);
}
