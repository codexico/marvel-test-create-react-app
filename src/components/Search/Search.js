import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { ReactComponent as SearchIcon } from '../../img/search.svg';

import './Search.css';

function Search({ handleSearch }) {
  const [value, setValue] = useState('');

  function handleSubmit(e) {
    e.preventDefault();
    if (value.trim()) {
      handleSearch(value.trim());
    }
  }

  function handleChange(e) {
    setValue(e.target.value);
  }

  return (
    <div className="Search">
      <form className="Search-form" onSubmit={handleSubmit}>
        <div className="Search-inputGroup">
          <label htmlFor="" className="Search-label">
            <button className="Search-inputButton" type="button">
              <SearchIcon className="Search-inputIcon" />
            </button>
            <input
              className="Search-input"
              placeholder="Procure por herÃ³is"
              value={value}
              onChange={handleChange}
              maxLength={200}
            />
          </label>
        </div>
      </form>
    </div>
  );
}

Search.propTypes = {
  handleSearch: PropTypes.func.isRequired,
};

export default Search;
