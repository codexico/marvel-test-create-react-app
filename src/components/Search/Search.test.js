import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Search from './Search';

test('Component renders correctly', () => {
  const { asFragment } = render(<Search handleSearch={() => {}} />);

  expect(asFragment()).toMatchSnapshot();
});

test('Component handles search', () => {
  const handleSearch = jest.fn();

  const { getByPlaceholderText } = render(
    <Search handleSearch={handleSearch} />
  );

  const input = getByPlaceholderText('Procure por her√≥is');

  fireEvent.change(input, { target: { value: '  Hulk  ' } });

  fireEvent.submit(input);

  expect(handleSearch).toHaveBeenCalledTimes(1);
  expect(handleSearch).toHaveBeenCalledWith('Hulk');

  fireEvent.change(input, { target: { value: '   ' } });
  fireEvent.submit(input);
  expect(handleSearch).toHaveBeenCalledTimes(1);
});
