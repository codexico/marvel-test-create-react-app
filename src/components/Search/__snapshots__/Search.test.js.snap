// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <form
      class="Search-form"
    >
      <div
        class="Search-inputGroup"
      >
        <label
          class="Search-label"
          for=""
        >
          <button
            class="Search-inputButton"
            type="button"
          >
            <svg
              class="Search-inputIcon"
            >
              search.svg
            </svg>
          </button>
          <input
            class="Search-input"
            maxlength="200"
            placeholder="Procure por herÃ³is"
            value=""
          />
        </label>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
